pkg/controller/networkpolicy/networkpolicy_controller.go

```diff
@@ -1225,6 +1348,72 @@ func podToMemberPod(pod *v1.Pod, includeIP, includePodRef bool) *controlplane.Gr
 	return memberPod
 }
 
+func externalEntityToGroupMember(ee *v1alpha1.ExternalEntity) *controlplane.GroupMember {
+	memberEntity := &controlplane.GroupMember{}
+	for _, endpoint := range ee.Spec.Endpoints {
+		var networkingPorts []controlplane.NamedPort
+		for _, port := range endpoint.Ports {
+			networkingPorts = append(networkingPorts, controlplane.NamedPort{
+				Port:     port.Port,
+				Name:     port.Name,
+				Protocol: controlplane.Protocol(port.Protocol),
+			})
+		}
+		ep := controlplane.Endpoint{
+			IP:    ipStrToIPAddress(endpoint.IP),
+			Ports: networkingPorts,
+		}
+		memberEntity.Endpoints = append(memberEntity.Endpoints, ep)
+	}
+	entityRef := controlplane.ExternalEntityReference{
+		Name:      ee.Name,
+		Namespace: ee.Namespace,
+	}
+	memberEntity.ExternalEntity = &entityRef
+	return memberEntity
+}
+
+func (n *NetworkPolicyController) processSelector(groupSelector antreatypes.GroupSelector) ([]*v1.Pod, []*v1alpha1.ExternalEntity) {
+	var pods []*v1.Pod
+	var externalEntities []*v1alpha1.ExternalEntity
+	if groupSelector.Namespace != "" {
+		// Namespace presence indicates Pods and ExternalEnitities must be selected from the same Namespace.
+		if groupSelector.PodSelector != nil {
+			pods, _ = n.podLister.Pods(groupSelector.Namespace).List(groupSelector.PodSelector)
+		} else if groupSelector.ExternalEntitySelector != nil {
+			externalEntities, _ = n.externalEntityLister.ExternalEntities(groupSelector.Namespace).List(groupSelector.ExternalEntitySelector)
+		}
+	} else if groupSelector.NamespaceSelector != nil && (groupSelector.PodSelector != nil || groupSelector.ExternalEntitySelector != nil) {
+		// Pods and ExternalEntities must be selected from Namespaces matching nsSelector.
+		namespaces, _ := n.namespaceLister.List(groupSelector.NamespaceSelector)
+		for _, ns := range namespaces {
+			if groupSelector.PodSelector != nil {
+				nsPods, _ := n.podLister.Pods(ns.Name).List(groupSelector.PodSelector)
+				pods = append(pods, nsPods...)
+			} else if groupSelector.ExternalEntitySelector != nil {
+				nsExtEntities, _ := n.externalEntityLister.ExternalEntities(ns.Name).List(groupSelector.ExternalEntitySelector)
+				externalEntities = append(externalEntities, nsExtEntities...)
+			}
+		}
+	} else if groupSelector.NamespaceSelector != nil {
+		// All the Pods and EEs from Namespaces matching the nsSelector must be selected.
```

Comment:
Is this correct? If I create a K8s NetworkPolicy that allows to all Pods in all Namespaces, both PodSelector and ExternalEntitySelector will be nil, but EEs shouldn't be selected?

Reply:
Thanks for spotting this. Discussed with @abhiraut and agreed the selector semantics should be compatible with K8s. So for any clause that has a namespaceSelector and nothing else, it will select all PODS in that namespace. If the intent is indeed to select all externalEntities in namespaces, user need to explicitly specify something like 
```
- namespaceSelector:
      matchLabels:
         project: myproject
   externalEntitySelector: {}
```
The code is updated accordingly. @suwang48404 FYI.